packages/grpc-js/src/metadata.ts

```diff
@@ -39,6 +42,7 @@ function normalizeKey(key: string): string {
 }
 
 function validate(key: string, value?: MetadataValue): void {
+  log(LogVerbosity.DEBUG, `validate key = "${key}" value type = "${typeof value}"`);
```

Comment:
Is it valid to use `printf()` style logging in gRPC? Since this defaults to using `console.error()`, it should at least work in that case - I'm just not sure if it's officially supported by gRPC.

If `printf()` style is supported, I'd recommend using it in this particular case. Template strings can get expensive, since they are evaluated even if nothing is actually logged (see today's Node security releases, which include a commit to move to fewer template strings).

Reply:
`log` isn't part of the grpc public API, so we can do whatever we want with it. The `setLogger` API documentation says that the logger should be a "`console`-like object", so I think it's reasonable to call it as though it was `console.error`.

Reply:
talking with @murgatroid99 offline, I'm a little concerned logging this might be a bit too noisy ... I think we'd do better to figure out some more targeted log lines that could help us start to figure out what the heck is going on when keys are corrupted.